compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7567,7 +7569,9 @@ private BLangClassDefinition desugarTemplateLiteralObjectTypedef(List<BLangLiter
         BLangFunction tempGeneratedInitFunction = createGeneratedInitializerFunction(classDef, env);
         tempGeneratedInitFunction.clonedEnv = SymbolEnv.createFunctionEnv(tempGeneratedInitFunction,
                                                                           tempGeneratedInitFunction.symbol.scope, env);
-        this.semanticAnalyzer.analyzeNode(tempGeneratedInitFunction, env);
+        SemanticAnalyzer.AnalyzerData data = new SemanticAnalyzer.AnalyzerData();
```

Comment:
Will it be better to provide `env` in the ctor. We can get rid of the default ctor as well.